### ISSUE TITLE
should pytest be in install_requires?

### ISSUE THREAD

**Issue by lafrech** [santoshphilip/eppy#123]
In setup.py:

```py
    tests_require=['pytest'],
    [...]
    install_requires = [
        "munch>=2.0.2",
        "beautifulsoup4>=4.2.1",
        "pytest>=2.3.5",
        "tinynumpy>=1.2.1",
        "six>=1.10.0",
        "decorator>=4.0.10"
        ],
    [...]
    extras_require={
        [...]
        'testing': ['pytest'],        
    }
```

I don't think `pytest` should appear in `install_requires`.

I was about to send a PR but I'm not sure whether it should be in `tests_require`, `extras_require['testing']` or both.

    tests_require=[
        'pytest>=2.3.5'
    ],

    extras_require={
        [...]
        'testing': [
            'pytest>=2.3.5'
        ],
    }

**Comment by jamiebull1** [comment]
I think I prefer it in `tests_require`, based on the logic [here](http://stackoverflow.com/a/32513360/1706564).

**Comment by lafrech** [comment]
The answer you point to rather suggests to put it in `extras_require`, doesn't it?

I don't mind and it may not be that important one way or another.

My point was that it did not have to be installed by default, so it shouldn't be in `install_requires`.

**Comment by jamiebull1** [comment]
On second reading, you're right. I think `extras_requires` is better. And certainly right that it shouldn't be in `install_requires`.

**Comment by jamiebull1** [comment]
In the end, I think `tests_require` integrates better with `setuptools` so I've gone with that in the PR I just raised (#129 which also addresses some other CI issues).

**Comment by jamiebull1** [comment]
Closing as implemented/addressed in #129
